templates/partials/messages_list.html

```diff
@@ -1,10 +1,10 @@
 <div id="message-list" class="col-sm-4 inbox-items left-pane">
   <ul ng-show="!loading || appending">
-    <li ng-repeat="message in messages | orderBy:'-date'" data-record-id="{{message.key}}" class="content-row" ng-class="{ 'selected': message.key === loadingContent || !loadingContent && message.key === selected.id, 'unread': !message.read }">
-      <a href="#messages/{{message.key}}">
+    <li ng-repeat="message in messages | orderBy:'-date'" data-record-id="{{message.key}}" data-record-type="{{message.type}}" class="content-row" ng-class="{ 'selected': message.key === loadingContent || !loadingContent && message.key === selected.id, 'unread': !message.read }">
+      <a href="#messages/{{message.type}}/{{message.key}}">
```

Comment:
The more I think about this route the more I think it should be: `{{message.type}}:{{message.key}}` (note the `:` instead of your `/`). The type alone makes no sense and will 404 because it's really just a qualifier for the key. This would also mean we could implement backwards compatibility because if there was no `:` then we can revert to the old logic. What do you think?

Reply:
Swapping `/` for `:` doesn't stop 404s. I've changed it though. It's not clear to me how to support two URL routes to the same state (it doesn't seem possible without having N states, which means we have to change more logic than I'd like) and I'm not sure it's worth the benefit.

Reply:
So I added a redirect that redirects `messages/{uuid}` to `messages/contact:{uuid}`, which works well enough (as long as the old URL they had was a contact.

If you want we can go all out and have a `messages/legacy:{uuid}` type or something, which actually works it out (a) attempts to pull the contact, then b) attempts to find the phone number in a view, then c) presumes it's unknown) if you want to be epic about it